app/models/concerns/taxonomix.rb

```diff
@@ -132,6 +132,44 @@ def scope_by_taxable_ids(scope)
         scope.where("#{table_name}.id IN (#{cached_ids.join(',')})")
       end
     end
+
+    def join_organizations
+      join_taxonomies(Organization.unscoped, org_join_arel)
+    end
+
+    def join_locations
+      join_taxonomies(Location.unscoped, loc_join_arel)
+    end
+
+    def join_taxonomies(taxonomies, tax_arel)
```

Comment:
```suggestion
    def join_taxonomies(tax_arel)
```

Reply:
I think the answer is yes. Otherwise both organizations and locations will be included in 1 join. For join_organizations we only want to join all organiztions. For join_locations, we only want to join all locations.